### ISSUE TITLE
Fix typescript typings after goog module conversion

### ISSUE THREAD

**Issue by alschmiedt** [google/blockly#5621]
**Describe the bug**
After the goog module conversion our typescript typings have been badly broken. We need to finish fixing these up before we do the Q4 release.

**Additional context**
@maribethb to add more context/information.

**Comment by rachel-fenichel** [comment]
Additional context/information:

For the past year+ we have used `typescript-closure-tools` to parse JSDoc and output `.d.ts` files. This toolchain broke when we converted to `goog.module`. 

We are moving to Typescript starting in Q1 2022, at which point we can just use the typescript compiler to make these files.

Between now and then, we're using a Rube Goldberg machine to generate `.d.ts` files. It involves auto-converting code to es6 modules on a best-effort basis and then using the Typescript compiler. We will publish the resulting files in the Q4 release, and may need to manually update them during Q1 2022 if developers find outstanding issues.
